.jenkins/publish.Jenkinsfile

```diff
@@ -0,0 +1,59 @@
+def releaseNoteBody
+def releaseVersion
+def releaseUrl
+
+pipeline {
+    libraries { lib 'adverity-shared-library@1.0.2' }
+    agent { label "PRJob" }
+    environment {
+        NODE_VERSION = getNodeVersion()
+    }
+    stages {
+        stage('Publish') {
+            environment {
+                GITHUB_TOKEN = credentials('e8c4eee6-cef3-4fd5-a65c-1050f7ecb0c7')
+                CURRENT_VERSION = "${readJSON(file: './package.json').version}"
+            }
+            stages {
+                stage('Changesets') {
+                    steps {
+                        script {
+                            releaseVersion = computeNextSemanticVersion(currentVersion: env.CURRENT_VERSION)
+                            releaseNoteBody = generateReleaseNote()
+                        }
+                    }
+                }
+                stage('Release') {
+                    steps {
+                        gitFlowRelease(version: releaseVersion) {
+                            sh "npm version ${releaseVersion} --no-git-tag-version"
+                            cleanChangesets()
+                        }
+                        script {
+                            releaseUrl = publishRelease(
+                                repository: 'pmedianetwork/eslint-config',
+                                tag: releaseVersion,
+                                body: releaseNoteBody
+                            )
+                        }
+                    }
+                }
+            }
+        }
+    }
+    post {
+        success {
+            slackSend(
+                color: 'good',
+                message: "Version ${releaseVersion} of eslint-config was released! :tada: (${releaseUrl})",
+                channel: 'dev-frontend'
+            )
+        }
+        unsuccessful {
+            notifySlack channel: 'dev-frontend'
+        }
+        cleanup {
+            cleanWs()
```

Comment:
what's `cleanWs`?

Reply:
cleans the workspace on the jenkins server. This is a little more resource heavier, since it needs to clone the repository again and stuff.. but since this workflow does some release stuff like merging an pushing, and because releasing doesn't happen that often. I chose to clean the workspace after every run so the state of git is always "new" when this pipeline is triggered